core/src/Kokkos_SYCL.hpp

```diff
@@ -134,7 +134,7 @@ class SYCL {
 
   static void impl_initialize(InitializationSettings const&);
 
-  int sycl_device() const;
+  static int sycl_device();
```

Comment:
Absolutely cannot be called `sycl_device` as it is misleading users.
One could expect it returns `sycl::device` just like `sycl_queue() -> sycl::queue`.
I don't think we need to expose any public member in the execution space class for it.
```suggestion
```
Instead do
```C++
template <>
struct DeviceTypeTraits<Kokkos::Experimental::SYCL> {
  static constexpr DeviceType id = DeviceType::SYCL;
  static int device_id(const Kokkos::Experimental::SYCL& exec) {
    return exec.impl_internal_space_instance().m_syclDev;
  }
};
```